### PR TITLE
`RecipeReader::get_value()` can now sub vars in strings

### PR DESCRIPTION
**Pull request by schuylermartin45**
- Adds support for the access operator (on strings) and the `upper` JINJA
  function
- Fixes several issues with JINJA variable subsitutions
- General code clean-up
- Updates tests with improvements

### CODE REVIEW

File: conda_recipe_manager/parser/recipe_reader.py [hunk 1]
```diff
@@ -225,6 +222,48 @@ def _generate_subtree(value: JsonType) -> list[Node]:
         # Primitives can be safely stringified to generate a parse tree.
         return RecipeReader(str(stringify_yaml(value)))._root.children  # pylint: disable=protected-access
 
+    def _set_on_schema_version(self) -> tuple[int, re.Pattern[str]]:
+        """
+        Helper function for `_render_jinja_vars()` that initialize `schema_version`-specific substitution details.
```
Comment (beeankha) [inline conda_recipe_manager/parser/recipe_reader.py:227]:
```suggestion
        Helper function for `_render_jinja_vars()` that initializes `schema_version`-specific substitution details.
```

File: conda_recipe_manager/parser/recipe_reader.py [hunk 2]
```diff
@@ -233,38 +272,35 @@ def _render_jinja_vars(self, s: str) -> JsonType:
...
             if key in self._vars_tbl:
                 # Replace value as a string. Re-interpret the entire value before returning.
                 value = str(self._vars_tbl[key])
                 if lower_match:
                     value = value.lower()
+                if upper_match:
+                    value = value.upper()
+                if idx_match:
+                    idx = int(cast(str, idx_match.group(2)))
+                    # From our research, it looks string indexing on JINJA variables is almost exclusively used to get
```
Comment (beeankha) [inline conda_recipe_manager/parser/recipe_reader.py:293]:
```suggestion
                    # From our research, it looks like string indexing on JINJA variables is almost exclusively used to get
```

**Review by beeankha — COMMENTED** [review]
Just some nitpicky docstring/comment wording changes, otherwise LGTM!

**Review by beeankha — APPROVED** [review]

**Comment by schuylermartin45** [comment]
Ah, we exceeded the line length on some of the suggestion reviews. This is what I get for merging at the local German brewery during Oktoberfest...

**Pull request merged**